### PR TITLE
Move payment calculations from onblock to claimrewards 

### PR DESCRIPTION
**Pull request by zorba80**

### CODE REVIEW

File: contracts/eosio.system/producer_pay.cpp [hunk 1]
```diff
@@ -120,34 +98,34 @@ namespace eosiosystem {
...
             p.last_rewards_claim = now();
             p.produced_blocks = 0;
          });
-      
-      INLINE_ACTION_SENDER(eosio::token, transfer)( N(eosio.token), {N(eosio),N(active)},
-                                                    { N(eosio), owner, rewards, std::string("producer claiming rewards") } );
-      
-                                                    */
+
+      INLINE_ACTION_SENDER(eosio::token, transfer)( N(eosio.token), {{N(eosio),N(active)}},
```
Comment (bytemaster) [inline contracts/eosio.system/producer_pay.cpp]: You should not transfer to self, "to savings" is just internal eosio.system state, no need for this last send.

File: contracts/eosio.system/producer_pay.cpp [hunk 2]
```diff
@@ -120,34 +98,34 @@ namespace eosiosystem {
...
+      const asset token_supply = token( N(eosio.token)).get_supply(symbol_type(system_token_symbol).name() );
+      const time time_since_last_fill = now() - parameters.last_pervote_bucket_fill;
+
+      const asset to_eos_bucket = supply_growth( standby_rate, token_supply, time_since_last_fill );
+      const asset to_savings    = supply_growth( continuous_rate - (perblock_rate + standby_rate), token_supply, time_since_last_fill );
+      const asset perblock_pay  = payment_per_block( perblock_rate, token_supply, prod->produced_blocks );
+      const asset issue_amount  = to_eos_bucket + to_savings + perblock_pay;
       
-      eosio_assert( rewards > asset(0, S(4,EOS)), "no rewards available to claim" );
+      INLINE_ACTION_SENDER(eosio::token, issue)( N(eosio.token), {{N(eosio),N(active)}},
```
Comment (bytemaster) [inline contracts/eosio.system/producer_pay.cpp]: the memo is wrong here, some of these tokens are to savings.

Comment (zorba80) [reply contracts/eosio.system/producer_pay.cpp]: Changed memo

File: contracts/eosio.system/producer_pay.cpp (hunk 1 above)
Comment (zorba80) [reply contracts/eosio.system/producer_pay.cpp]: I deleted the transfer.

**Pull request merged**